internal/services/extendedlocation/extended_location_custom_location.go

```diff
@@ -0,0 +1,348 @@
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"namespace": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"cluster_extension_ids": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+
+		"host_resource_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"authentication": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"type": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"value": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"display_name": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"host_type": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(customlocations.HostTypeKubernetes),
+			}, false),
+		},
+	}
+}
+
+func (r CustomLocationResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r CustomLocationResource) ModelObject() interface{} {
+	return &CustomLocationResourceModel{}
+}
+
+func (r CustomLocationResource) ResourceType() string {
+	return "azurerm_extended_custom_location"
+}
+
+func (r CustomLocationResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model CustomLocationResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+
+			id := customlocations.NewCustomLocationID(subscriptionId, model.ResourceGroupName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			customLocationProps := customlocations.CustomLocationProperties{}
+
+			if model.Authentication != nil {
+				auth := model.Authentication[0]
+				customLocationProps.Authentication = &customlocations.CustomLocationPropertiesAuthentication{
+					Type:  &auth.Type,
+					Value: &auth.Value,
+				}
```

Comment:
I think we need to check the length before accessing the index 0, because it could be an empty block in the config.

Reply:
Will fix.